website/docs/secrets/managing-secrets-with-sops.mdx

```diff
@@ -0,0 +1,227 @@
+
+```bash
+rm -f age.ageKey
+```
+
+5- Create a kustomization for reconciling the secrets on the cluster and set the `--decryption-secret` flag to the name of the private key created in step 2
+
+```bash
+flux create kustomization age-secrets \
+--source=secrets \ # the git source to reconcile the secrets from
+--path=./secrets/age \
+--prune=true \
+--interval=10m \
+--decryption-provider=sops \
+--decryption-secret=sops-age-private-key
+```
+
+6- Annotate the kustomization object created in the previous step with the name and namespace of the public key created in step 4.
+
+```bash
+kubectl annotate kustomization secrets sops-public-key/name=sops-age-public-key sops-public-key/namespace=flux-system
+```
+
+The kustomization object should looks like this
+
+```yaml
+apiVersion: kustomize.toolkit.fluxcd.io/v1beta2
+kind: Kustomization
+metadata:
+  name: age-secrets
+  namespace: flux-system
+  annotations:
+    sops-public-key/name: sops-age-public-key
+    sops-public-key/namespace: flux-system
+spec:
+  interval: 10m
+  sourceRef:
+    kind: GitRepository
+    name: secrets
+  path: ./secrets/age
+  decryption:
+    provider: sops
+    secretRef:
+      name: sops-age-private-key
+  prune: true
+  validation: server
+```
+
+> This is an essential step in order to allow other operators and developers to utilize WeaveGitOps UI to encrypt SOPS secrets using the public key secret in the cluster.
+
+> In case of using OpenPGP and age in the same cluster, you need to make the kustomizations point to different directories. This is because flux's kustomize-controller expects that all the secrets in the kustomization's path are encrypted with the same key.
+
+
+## Security Recommendations
+
+Access to sops decryption secrets should be restricted and allowed only to be read by flux's kustomize controller. This can be done using Kubernetes RBAC.
+
+Here's an example of how you can use RBAC to restrict access to sops decryption secrets:
+
+1. Create a new Kubernetes role that grants read access to sops decryption secrets
+
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: Role
+metadata:
+  name: sops-secrets-role
+rules:
+- apiGroups: [""]
+  resources: ["secrets"]
+  resourceNames: ["sops-gpg-private-key", "sops-age-private-key"]
+  verbs: ["get", "watch", "list"]
+```
+
+2. Bind the role to the service account of the flux's kustomize-controller
+
+```yaml
+apiVersion: rbac.authorization.k8s.io/v1
+kind: RoleBinding
+metadata:
+  name: sops-secrets-rolebinding
+roleRef:
+  apiGroup: rbac.authorization.k8s.io
+  kind: Role
+  name: sops-secrets-role
+subjects:
+- kind: ServiceAccount
+  name: kustomize-controller
+```
+
+> You would need to ensure that no other rolebindings or clusterrolebndings would allow read the the decryption secret at any time. This could be achieved by leveraging policy capabilities to detect existing and prevent future creation of roles that would grant read secrets permissions. 
```

Comment:
```suggestion
> You would need to ensure that no other rolebindings or clusterrolebndings would allow reading the the decryption secret at any time. This could be achieved by leveraging policy capabilities to detect existing and prevent future creation of roles that would grant read secrets permissions. 
```